### PR TITLE
Initial implementation of the base platform

### PR DESCRIPTION
**Pull request by arueth**

### CODE REVIEW

File: platforms/gke/base/_shared_config/versions.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+
+terraform {
+  required_version = ">= 1.5.7"
+
+  required_providers {
+  }
+
+  provider_meta "google" {
+    module_name = "cloud-solutions/ml-platform-cluster-shared_config-deploy-v1"
```
Comment (laurentgrangeau) [inline platforms/gke/base/_shared_config/versions.tf:22]: Is this really ML platform or accelerated-platform to keep track of the deployment ?

File: platforms/gke/base/_shared_config/workloads_variables.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Configuration dependencies
+#
+
+locals {
+  manifests_directory = abspath("${path.module}/../manifests")
+}
+
+variable "kueue_version" {
+  default = "0.8.0"
```
Comment (laurentgrangeau) [inline platforms/gke/base/_shared_config/workloads_variables.tf:24]: In the batch ref repo, kueue version is 0.8.1. I think we should harmonise versions

File: platforms/gke/base/core/gke_enterprise/configmanagement/git/feature.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  fleet_default_member_config {
+    configmanagement {
+      management = "MANAGEMENT_AUTOMATIC"
+
+      config_sync {
+        enabled       = true
+        source_format = "unstructured"
+
+        # git {
```
Comment (laurentgrangeau) [inline platforms/gke/base/core/gke_enterprise/configmanagement/git/feature.tf:30]: Remove unnecessary comments

File: platforms/gke/base/core/gke_enterprise/policycontroller/feature.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+resource "google_gke_hub_feature_membership" "cluster_spolicycontroller" {
+  provider = google-beta
+
+  feature    = google_gke_hub_feature.policycontroller.name
+  location   = google_gke_hub_feature.policycontroller.location
+  membership = data.google_container_cluster.cluster.name
+  project    = data.google_project.default.project_id
+
+  policycontroller {
+    #enabled                    = true
```
Comment (laurentgrangeau) [inline platforms/gke/base/core/gke_enterprise/policycontroller/feature.tf:32]: Remove unnecessary comments

File: platforms/gke/base/core/workloads/kueue/dashboards/kueue-monitoring-dashboard.json
Comment (laurentgrangeau) [inline platforms/gke/base/core/workloads/kueue/dashboards/kueue-monitoring-dashboard.json:1]: Does it include the PR from James ?
https://github.com/GoogleCloudPlatform/ai-on-gke/pull/856

File: test/ci-cd/cloudbuild/platforms-gke-base-core-terraform.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      platforms/gke/base/core/teardown.sh || exit 1
+      if [[ $${APPLY_RETURN_CODE} -ne 0 ]]; then
+        exit $${APPLY_RETURN_CODE}
+      fi
+  waitFor:
+  - "Build runner image"
+
+substitutions:
+  _AR_REPO_LOCATION: "us-central1"
+  _IAP_DOMAIN: "accelerated-platforms.joonix.net"
```
Comment (laurentgrangeau) [inline test/ci-cd/cloudbuild/platforms-gke-base-core-terraform.yaml:53]: Maybe tell people to substitute with their own domain. Is there somewhere an explanation on how to setup IAP for this solution ?

**Review by laurentgrangeau — CHANGES_REQUESTED** [review]
Some minor changes and it should be good to go. At some point we should thing about defining a structure in the variable for defining node in the nodepool as there is lots of duplicate code that can be refactor to a simple TF script with a foreach loop.

File: platforms/gke/base/_shared_config/versions.tf (hunk 1 above)
Comment (arueth) [reply platforms/gke/base/_shared_config/versions.tf:22]: Yeah, I changed the format of the string and must have missed that one.

File: platforms/gke/base/core/gke_enterprise/configmanagement/git/feature.tf (hunk 3 above)
Comment (arueth) [reply platforms/gke/base/core/gke_enterprise/configmanagement/git/feature.tf:30]: Those Terraservices are still under development. I'm just trying to get the scaffolding in so other could work on it if they want.

File: platforms/gke/base/core/gke_enterprise/policycontroller/feature.tf (hunk 4 above)
Comment (arueth) [reply platforms/gke/base/core/gke_enterprise/policycontroller/feature.tf:32]: That Terraservices are still under development. I'm just trying to get the scaffolding in so other could work on it if they want.

File: platforms/gke/base/core/workloads/kueue/dashboards/kueue-monitoring-dashboard.json
Comment (arueth) [reply platforms/gke/base/core/workloads/kueue/dashboards/kueue-monitoring-dashboard.json:1]: Yes, it includes the GMP and dashboard.

File: test/ci-cd/cloudbuild/platforms-gke-base-core-terraform.yaml (hunk 5 above)
Comment (arueth) [reply test/ci-cd/cloudbuild/platforms-gke-base-core-terraform.yaml:53]: So these are currently only used for our internal CI/CD on the repo. If we decide to make a version for public consumption I will create a README for it.

**Comment by arueth** [comment]
> Some minor changes and it should be good to go. At some point we should thing about defining a structure in the variable for defining node in the nodepool as there is lots of duplicate code that can be refactor to a simple TF script with a foreach loop.

Yes, I've tried to keep the code as WET as possible since it is for reference purposes. The node pools are really just for reference purposes. Other solutions will more than likely have their own node pool requirements so they will probably skip the container_node_pool Terraservice and implement their own. That is the intended use of the `platforms/gke/base/use-cases`  and `platforms/gke/base/features`  folders.

**Review by laurentgrangeau — APPROVED** [review]
LGTM

**Pull request merged**